e3db/src/main/java/com/tozny/e3db/Client.java

```diff
@@ -1389,11 +1450,23 @@ public void run() {
    * @param eakInfo The key to use for encrypting.
    * @return The deccrypted record.
    */
-  public Record decryptExisting(Record record, EAKInfo eakInfo) {
+  public Record decryptExisting(Record record, EAKInfo eakInfo) throws E3DBException {
+    if (eakInfo.getSignerSigningKey().isEmpty()) {
+      throw new E3DBException("EAKInfo has no signing key");
+    }
+
     byte[] ak = getCachedAccessKey(record, eakInfo);
 
     try {
       Map<String, String> plainRecord = decryptObject(ak, record.data());
+
+      Record decrypted = new LocalRecord(plainRecord, record.meta(), record.signature());
+
+      SignedDocument<Record> document = sign((Record) new LocalRecord(plainRecord, record.meta()));
```

Comment:
This bit of code is confusing. Are you signign the record again and then verifying that? 

Also the `Record` cast should be unnecessary.